bioconda_utils/build.py

```diff
@@ -63,10 +63,22 @@ def build(recipe,
         If mulled_test == True, the name of the mulled docker image will be
         appended to this list.
     """
-    env = dict(env)
+
+    # Clean provided env and exisiting os.environ to only allow whitelisted env
+    # vars
+    _env = {}
+    _env.update({k:str(v) for k, v in os.environ.items() if utils.allowed_env_var(k)})
+    _env.update({k:str(v) for k, v in dict(env).items() if utils.allowed_env_var(k)})
```

Comment:
It's true that `env` should be coming from the `env_matrix` file and that shouldn't contain any secrets. But it is technically possible to pass `env=os.environ` to this function, so I was trying to guard against that. I don't think this hurts anything as-is, but I'm fine either way.